doc: clarify http.request supports headers as array

Dinesh0007000 opened the pull request:
This PR updates the documentation for `http.request()` to clarify that the `headers` option can also be provided as an array of key-value pairs, like `['Content-Type', 'text/plain', 'X-Custom', 'yes']`.

This behavior aligns with `response.writeHead()`, but was previously undocumented in the `http.request()` API docs.

Fixes: https://github.com/nodejs/node/issues/57986

<!--
Before submitting a pull request, please read:

- the CONTRIBUTING guide at https://github.com/nodejs/node/blob/HEAD/CONTRIBUTING.md
- the commit message formatting guidelines at
  https://github.com/nodejs/node/blob/HEAD/doc/contributing/pull-requests.md#commit-message-guidelines

For code changes:
1. Include tests for any bug fixes or new features.
2. Update documentation if relevant.
3. Ensure that `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes.

If you believe this PR should be highlighted in the Node.js CHANGELOG
please add the `notable-change` label.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

Dinesh0007000 committed:
doc: clarify http.request supports headers as array

This PR updates the documentation for `http.request()` to clarify that the `headers` option can also be an array of key-value pairs, similar to `response.writeHead`.

Fixes: https://github.com/nodejs/node/issues/57986

pimterry reviewed (changes requested):
Thanks for contributing @Dinesh0007000! This is indeed officially supported and it's great to get this documented.

There's a few issues here to tidy up to get this ready to merge. I've added a comment on the markdown, and the commit message check is also failing due to line length. The guidelines for commit messages are here: https://github.com/nodejs/node/blob/HEAD/doc/contributing/pull-requests.md#commit-message-guidelines

Dinesh0007000 committed:
doc: clarify headers array format in http.request options

Added a detailed explanation for how the 'headers' option in http.request
can also be passed as a flat array of alternating header names and values.

This provides clarity for beginners and aligns with the format used in
response.writeHead() and request.rawHeaders.

pimterry reviewed (changes requested):
There's a couple of tweaks required here, and the linting is still failing for the commit message & markdown. You'll need to look at the failure details there and fix this. For the markdown, you can test this locally with `make lint-md`.

For the commit message, the issue is that you've added a new commit, but the linting always runs against the first commit in the PR (because that's the commit that typically later appears in the changelog). You'll need to edit that previous commit instead and force push your change to overwrite it.

Dinesh0007000 committed:
fix: address markdown linting issues and link formatting

- Removed duplicate definition of [`response.writeHead()`] as it already exists earlier in the file.
- Updated [`request.rawHeaders`] link to be relative (#requestrawheaders) instead of using an absolute URL.

These fixes resolve markdown linting issues flagged by the CI and ensure consistency with documentation standards.

Dinesh0007000 reviewed (commented):
## fix: address markdown linting issues and link formatting
- Removed duplicate definition of [`response.writeHead()`] as it already exists earlier in the file.
- Updated [`request.rawHeaders`] link to be relative (#requestrawheaders) instead of using an absolute URL.

These fixes resolve markdown linting issues flagged by the CI and ensure consistency with documentation standards.

pimterry reviewed (changes requested):
Hi @Dinesh0007000, the markdown & commit message linting is still failing, and the markdown build is failing too. You'll need to run that and ensure it passes locally (both `make doc-only` and `make lint-md`) and you'll need to squash the commits & force push to fix the original commit message here before this will pass. Please make sure these commands both pass locally before pushing more changes.
